lib/Serialization/Serialization.cpp

```diff
@@ -738,8 +738,23 @@ void Serializer::writeHeader(const SerializationOptions &options) {
         options_block::XCCLayout XCC(Out);
 
         SDKPath.emit(ScratchRecord, M->getASTContext().SearchPathOpts.SDKPath);
-        for (const std::string &arg : options.ExtraClangOptions) {
-          XCC.emit(ScratchRecord, arg);
+        auto &Opts = options.ExtraClangOptions;
+        for (auto Arg = Opts.begin(), E = Opts.end(); Arg != E; ++Arg) {
+          // Filter out any -ivfsoverlay options that include an
+          // unextended-module-overlay.yaml overlay. By convention the Xcode
+          // buildsystem uses these while *building* mixed Objective-C and Swift
+          // frameworks; but they should never be used to *import* the module
+          // defined in the framework.
+          std::string Prefix = "-ivfsoverlay";
+          std::string Suffix = "unextended-module-overlay.yaml";
+          if (Prefix.compare(0, Prefix.length(), *Arg) == 0) {
+            auto Next = std::next(Arg);
+            if (Next != E && Suffix.compare(0, Suffix.length(), *Next)) {
+              ++Arg;
+              break;
```

Comment:
Isn't this skipping all subsequent options, and thus only correct if this was the last one?